src/main/java/com/esri/samples/map/generate_offline_map_with_local_basemap/GenerateOfflineMapWithLocalBasemap.java

```diff
@@ -0,0 +1,331 @@
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol;
+import com.esri.arcgisruntime.tasks.offlinemap.GenerateOfflineMapJob;
+import com.esri.arcgisruntime.tasks.offlinemap.GenerateOfflineMapParameters;
+import com.esri.arcgisruntime.tasks.offlinemap.GenerateOfflineMapResult;
+import com.esri.arcgisruntime.tasks.offlinemap.OfflineMapTask;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.geometry.Insets;
+import javafx.geometry.Point2D;
+import javafx.geometry.Pos;
+import javafx.scene.Scene;
+import javafx.scene.control.Alert;
+import javafx.scene.control.Button;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.ProgressBar;
+
+import javafx.scene.layout.StackPane;
+import javafx.scene.text.TextAlignment;
+import javafx.stage.Stage;
+import org.apache.commons.io.FilenameUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+public class GenerateOfflineMapWithLocalBasemap extends Application {
+
+  private MapView mapView;
+  private String downloadedBasemapSavedLocation;
+  private String localBasemapDirectory;
+  private File localBasemapFile;
+  private GraphicsOverlay graphicsOverlay;
+  private Graphic downloadArea;
+  private ArcGISMap map;
+
+  private ProgressBar progressBar;
+  private Button offlineMapButton;
+
+  private GenerateOfflineMapParameters generateOfflineMapParameters;
+  private GenerateOfflineMapJob generateOfflineMapJob;
+  private OfflineMapTask task;
+
+
+  @Override
+  public void start(Stage stage) throws Exception {
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+      scene.getStylesheets().add(getClass().getResource("/css/style.css").toExternalForm());
+
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Generate Offline Map With Local Basemap Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create a map view
+      mapView = new MapView();
+
+      // create a button to take the map offline
+      createTakeMapOfflineButton();
+
+      // create a portal item with the itemId of the web map
+      Portal portal = new Portal("https://www.arcgis.com");
+      PortalItem portalItem = new PortalItem(portal, "acc027394bc84c2fb04d1ed317aac674");
+
+      // create a map with the portal item
+      map = new ArcGISMap(portalItem);
+      addMapLoadedListener();
+      // set the map to the map view
+      mapView.setMap(map);
+      // update the download area box whenever the viewpoint changes
+      mapView.addViewpointChangedListener(viewpointChangedEvent -> updateDownloadArea());
+
+      // create a progress bar to show download progress
+      createProgressIndicator();
```

Comment:
Generally, you shouldn't use methods to initialize UI. You can either move the code inline here, or use FXML.